openupgrade_scripts/scripts/website/16.0.1.0/end-migration.py

```diff
@@ -1,38 +1,62 @@
 from openupgradelib import openupgrade
-from openupgradelib.openupgrade_160 import convert_string_bootstrap_4to5
-from psycopg2.extras import Json
+from openupgradelib.openupgrade_160 import (
+    _convert_field_bootstrap_4to5_sql,
+    convert_field_bootstrap_4to5,
+)
 
 
-def boostrap_5_migration(env):
+def convert_custom_qweb_templates_bootstrap_4to5(env):
     """Convert customized website views to Bootstrap 5."""
     backup_column = openupgrade.get_legacy_name("arch_db_bs4")
     openupgrade.logged_query(
         env.cr, f"ALTER TABLE ir_ui_view ADD COLUMN {backup_column} TEXT"
     )
+    openupgrade.logged_query(
+        env.cr,
+        f"""
+        UPDATE ir_ui_view SET {backup_column} = arch_db
+        WHERE type = 'qweb' AND website_id IS NOT NULL
+        """,
+    )
     # Find views to convert
     env.cr.execute(
-        """
-        SELECT iuv.id, iuv.arch_db
-        FROM ir_ui_view iuv
-        WHERE iuv.type = 'qweb'
-        """
+        "SELECT id FROM ir_ui_view WHERE type = 'qweb' AND website_id IS NOT NULL"
```

Comment:
Why only applying to those with `website_id` assigned? I thought on this too, but I finally decided to apply it to all the existing templates.

Reply:
Well, templates without website_id will be noupdate=0 records so it would be useless to update them or am I missing a case where there could be such custom views?

Reply:
Not sure if there can be any manually created view. If you are sure that it couldn't, then it's OK to continue this way.

Reply:
I won't say I'm 100% sure but I can't figure out any exception :sweat_smile:

Reply:
OK, let's keep it this way for now.